modules/services/ogc-api-records/src/main/java/org/fao/geonet/ogcapi/records/controller/model/CollectionInfo.java

```diff
@@ -10,45 +17,261 @@
 import javax.xml.bind.annotation.XmlAccessType;
 import javax.xml.bind.annotation.XmlAccessorType;
 import javax.xml.bind.annotation.XmlRootElement;
+import org.fao.geonet.ogcapi.records.model.OgcApiContact;
+import org.fao.geonet.ogcapi.records.model.OgcApiExtent;
+import org.fao.geonet.ogcapi.records.model.OgcApiLanguage;
+import org.fao.geonet.ogcapi.records.model.OgcApiLink;
+import org.fao.geonet.ogcapi.records.model.OgcApiTheme;
 
 /**
- * CollectionInfo entity.
+ * cf. https://github.com/opengeospatial/ogcapi-records/blob/master/core/openapi/schemas/catalog.yaml
+ *
+ * <p>CollectionInfo entity.
  */
 @JacksonXmlRootElement(localName = "CollectionInfo")
 @XmlRootElement(name = "CollectionInfo")
 @XmlAccessorType(XmlAccessType.FIELD)
-public class CollectionInfo   {
+public class CollectionInfo {
+
+  //required cf. https://github.com/opengeospatial/ogcapi-records/blob/master/core/openapi/schemas/catalog.yaml
+  @JsonProperty("type")
+  @JacksonXmlProperty(localName = "type")
+  private String type = "catalog";
+
+  //cf. https://github.com/opengeospatial/ogcapi-records/blob/master/core/openapi/schemas/catalog.yaml
+  // NOTE: spec says this should be "record" or ["record","catalog"].
+  @JsonProperty("itemType")
+  @JacksonXmlProperty(localName = "itemType")
+  private String itemType = "record";
+
+
+  /**The identifier of the catalog.*/
   @JsonProperty("id")
   @JacksonXmlProperty(localName = "id")
   private String id;
 
+  /**A human-readable name given to the resource.*/
   @JsonProperty("title")
   @JacksonXmlProperty(localName = "title")
   private String title;
 
+  /**A free-text account of the resource.*/
   @JsonProperty("description")
   @JacksonXmlProperty(localName = "description")
   private String description;
 
+  /**links for this object.*/
   @JsonProperty("links")
   @JacksonXmlProperty(localName = "links")
-  
-  private List<Link> links = new ArrayList<>();
+  @JsonInclude(Include.NON_EMPTY)
+  private List<OgcApiLink> links = new ArrayList<>();
 
+  /**The spatiotemporal coverage of this catalog.*/
   @JsonProperty("extent")
   @JacksonXmlProperty(localName = "extent")
-  private Extent extent;
+  private OgcApiExtent extent;
 
+  /**The list of supported coordinate reference systems.*/
   @JsonProperty("crs")
   @JacksonXmlProperty(localName = "crs")
-  
+  @JsonInclude(Include.NON_EMPTY)
   private List<String> crs = null;
 
+  /** A list of contacts qualified by their role(s) in association
```

Comment:
It's missing `recordLanguages`, but not sure how relevant it is. See https://docs.ogc.org/DRAFTS/20-004.html#collection-properties-table

Reply:
I will add this.  I also see that there are a few others in the written spec versus whats in the .yml.

Reply:
I added the extra object - the "Schema" object is VERY VERY complex.  I didn't fully include it (because of the pattern properties).  I've left TODOs for when this is necessary (its not at the moment).  I notice that pygeoapi seems to include this.